### ISSUE TITLE
Docs suggestion: clarify terminology of formatting/style/logic

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint-stylistic/eslint-stylistic#19]
Surfacing https://github.com/eslint/eslint.org/issues/419 here: in my experience, many users have a hard time differentiating between the concepts of _"formatting"_ and _"style"_. It doesn't help that different ecosystems or tools within the same ecosystem can have subtle differences. I personally had a different vocab from what ESLint core uses until this year!

My understanding of what ESLint core uses -and therefore typescript-eslint too- is what's on https://typescript-eslint.io/linting/troubleshooting/formatting#eslint-core-and-formatting right now:

> Most lint rules fall into one of two to three categories:
> 
> - **Logical**: Rules that care about the logic in runtime behavior of code (such as missing `await`s or invalid logical checks).
> - **Stylistic**: Rules that care about style concerns which do impact runtime behavior of code, but generally not logic. These are mostly around naming or which roughly-equivalent syntax constructs to use (such as function declarations vs. arrow functions).
>   - **Formatting**: Stylistic rules that care only about trivia (semicolons, whitespace, etc.) without impacting the runtime behavior of the code. These rules conflict with dedicated formatters such as Prettier.

In other words: the two main categories are _logical_ and _stylistic_, and _formatting_ is a subset of _stylistic_.

It makes sense to call this repo & org _eslint-**stylistic**-_ given that the plugin is likely to eventually include some stylistic rules alongside the many formatting ones. But it'd help users understand that if the docs could mention that explicitly. Especially since many users are using eslint-stylistic primarily for _formatting_ concerns.

_(this issue came out of a conversation with @antfu, who explained that last paragraph - thanks!)_

Aside: [eslint.style](https://eslint.style) right now (before full release) has _much_ higher quality docs than most projects do on or after release. Love it. Nicely done! ❤️

**Comment by antfu** [comment]
Thanks for bringing this up. I have written some that basically align with TypeScript ESLint's terminology. Let me know if you think anything can be improved. Thanks.

**Comment by JoshuaKGoldberg** [comment]
Super, thanks! I left a few small comments on the commit, nothing major. Looks great! 🚀
